### DISCUSSION TITLE
Caught OS signal, shutting down

### DISCUSSION THREAD

**Discussion by a-camacho** [qdm12/ddns-updater#305] [Q&A]
Hi guys,

I have a weird problem since I changed my Internet Router.
DDNS Updater seems to work, but at the end the container stops ... and recreates itself, as I'm using a Docker Stack.

Before it seems to me that the service just stayed online. Checking every X time.

Is it something you saw already ?
Why is the container telling me `Caught OS signal, shutting down`?

Thanks a lot !

```
Running version latest built on 2022-01-21T13:37:49Z (commit f5b76f2)

🔧 Need help? https://github.com/qdm12/ddns-updater/discussions/new
🐛 Bug? https://github.com/qdm12/ddns-updater/issues/new
✨ New feature? https://github.com/qdm12/ddns-updater/issues/new
☕ Discussion? https://github.com/qdm12/ddns-updater/discussions/new
💻 Email? quentin.mcgaw@gmail.com
💰 Help me? https://www.paypal.me/qmcgaw https://github.com/sponsors/qdm12
2022/02/04 10:24:00 INFO reading JSON config from file /updater/data/config.json
2022/02/04 10:24:00 INFO Found 5 settings to update records
2022/02/04 10:24:00 INFO Reading history from database: domain {removed} host @
2022/02/04 10:24:00 INFO Reading history from database: domain {removed} host {removed}
2022/02/04 10:24:00 INFO Reading history from database: domain {removed}host {removed}
2022/02/04 10:24:00 INFO Reading history from database: domain {removed} host {removed}
2022/02/04 10:24:00 INFO Reading history from database: domain {removed} host {removed}
2022/02/04 10:24:00 INFO healthcheck server: listening on 127.0.0.1:9999
2022/02/04 10:24:00 INFO backup: disabled
2022/02/04 10:24:00 INFO http server: listening on :8000
2022/02/04 10:24:21 WARN Caught OS signal, shutting down
2022/02/04 10:24:21 WARN healthcheck server: shutting down (context canceled)
2022/02/04 10:24:21 WARN http server: shutting down (context canceled)
2022/02/04 10:24:21 WARN http server: shut down
2022/02/04 10:24:21 INFO Shutdown successful
```

**Comment by qdm12** [comment]
That means something external to the container sends it a sigterm or sigkill signal. That's either the docker daemon or the OS telling the process to shutdown. Maybe are you running out of memory?

**Reply by a-camacho** [reply]
Oh thanks for your reply.

That is possible, my Docker Host is a Proxmox Debian VM. It is quite under load, I have a lot of services running and my RAM usage is `95.55% (17.20 GiB of 18.00 GiB)` ... That could be the reason.

But weird that none of the rest of my Docker services are stopped because of low ram.
Don't really know If there is anything I can do - that is not adding more RAM or removing services.

**Reply by qdm12** [reply]
That's up for the OS to kill a certain process instead of another one, although usually it kills the one using the most memory. Let me know if you see some odd memory pattern for DDNS updater, although it should really use barely no memory (haven't checked but my guess is ~20MB or so).

**Comment by Danne-J** [comment]
i can see the same behavior.
my is for sure not running out of memory.

one thing i also notice is that the container not indicate as started for real, just starting.
it doing things and working updating DNSs and so forth.

wonder if it has something to do with the built in healtcheck that in my case docker swarm picking up and it feels like its not reporting back "OK"

**Reply by a-camacho** [reply]
Any clue regarding this ? I am still having this problem, and have now free RAM avalaible.

**Reply by qdm12** [reply]
You can try disabling the healthcheck with `--no-healthcheck` (see the docker run reference on healthcheck). But yes it's a signal coming from outside.

**Reply by Danne-J** [reply]
hi can confirm

```yml
    healthcheck:
      disable: true
```

docker-compose / stack will set the container in running

**Reply by qdm12** [reply]
@ic3cool actually do you see your container getting killed by a signal?

Seeing it in starting status for a few minutes at start is perfectly normal.

**Reply by Danne-J** [reply]
befor the container never exit start state to running and got killed by a signal.
Now its been in running state 17h and not been killed once, also i can see in the logs that DNS records are updated by intervall.

so i would say that the in container included default check is some what broken. Or need something else from deployment point of view added to work.

**Reply by Danne-J** [reply]
> @ic3cool actually do you see your container getting killed by a signal?
> 
> Seeing it in starting status for a few minutes at start is perfectly normal.

sorry missed to answer this.
befor i hade the same issue as @a-camacho

**Reply by qdm12** [reply]
Do you run any other container to restart unhealthy containers perhaps?
